warp10/src/main/java/io/warp10/script/functions/FIND.java

```diff
@@ -700,8 +702,14 @@ private Map<String,Object> paramsFromMap(Map<String,Object> map) throws WarpScri
 
   public static MetadataIterator getScopedIterator(DirectoryClient directoryClient, ReadToken rtoken, DirectoryRequest drequest) throws WarpScriptException, IOException {
     if (rtoken.getAttributesSize() > 0 && rtoken.getAttributes().containsKey(Constants.TOKEN_ATTR_SCOPE)) {
-      Object[] clslbls = PARSESELECTOR.parse(rtoken.getAttributes().get(Constants.TOKEN_ATTR_SCOPE));
 
+      Object[] clslbls;
+      try {
+        clslbls = PARSESELECTOR.parse(rtoken.getAttributes().get(Constants.TOKEN_ATTR_SCOPE));
+      } catch (WarpScriptException e) {
+        throw new WarpScriptException("invalid syntax for scope selector within the token.");
```

Comment:
Please change message to

"Invalid syntax for token scope selector."

Ditto below.